infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java

```diff
@@ -154,23 +155,23 @@ public int getConflictHashCode(UUID streamID, int conflictParam) {
     }
 
     /**
-     * Returns true if the txn commits.
      * If the request submits a timestamp (a global offset) that is less than one of the
      * global offsets of a streams specified in the request, then abort; otherwise commit.
      *
-     * @param txData info provided by corfuRuntime for conflict resolultion:
+     * @param txInfo info provided by corfuRuntime for conflict resolultion:
      *              - timestamp : the snapshot (global) offset that this TX reads
      *              - conflictSet: conflict set of the txn.
      *                if any conflict-param (or stream, if empty) in this set has a later timestamp than the snapshot, abort
+     *
+     * @return      Returns the type of token reponse based on whether the txn commits, or the abort cause.
      */
-    public boolean txnCanCommit(TxResolutionInfo txData) {
-        log.trace("txn[{}] resolution req snapshot version={}, conflicts={}",
-                txData.getTXid(),
-                txData.getSnapshotTimestamp(), txData.getConflictSet());
-
-        AtomicBoolean commit = new AtomicBoolean(true);
-        for (Map.Entry<UUID, Set<Integer>> entry : txData.getConflictSet().entrySet()) {
-            if (!commit.get())
+    public TokenType txnCanCommit(TxResolutionInfo txInfo) {
+        log.trace("Commit-req[" + txInfo + "]");
+
+        AtomicInteger response = new AtomicInteger(TokenType.NORMAL.ordinal());
```

Comment:
I think what you need is an AtomicReference, that way you do not have to deal with the ordinal of the tokentype.